### DISCUSSION TITLE
[Question] Is it possible to focus the text field of a SearchBar programatically?

### DISCUSSION THREAD

**Discussion by chaotic-dev** [flet-dev/flet#3350] [Q&A]
### Question

I have a `SearchBar` with a `ListView` populated with search suggestions as its control. When I click one of the `ListItem`s, it modifies the `SearchBar`'s `value` with the search suggestion. After doing so, the `SearchBar`'s input loses focus and I'd like it to remain focused for the user to continue typing. Is it possible to focus the `SearchBar`'s input or otherwise get its underlying `TextField` which has a focus method? I've attempted to set `autofocus` to `True` and call its `open_view` method to try and grab focus, but neither option seem to focus the input.

Thanks in advance

### Code sample

```python
# Stripped-down example
search_bar = ft.SearchBar(
    controls=[
        ft.ListTile(title=ft.Text("suggestion"), on_click=apply_suggestion, data="suggestion")
    ],
    on_submit=lambda _: page.go(f"/search/{search_bar.value}"),
    on_change=update_suggestions,
    autofocus=True,
)

def apply_suggestion(e):
    search_bar.value = e.control.data
    search_bar.update()
    search_bar.open_view()
```

Update:
OS information: Pop OS
Flet package version: 0.22.1

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Like this maybe?:
```python
import flet as ft


def main(page):
    page.horizontal_alignment = ft.CrossAxisAlignment.CENTER

    def close_anchor(e):
        text = f"Color {e.control.data}"
        anchor.value = text
        anchor.update()
        # print(f"closing view from {text}")
        # anchor.close_view(text)

    anchor = ft.SearchBar(
        view_elevation=4,
        divider_color=ft.colors.AMBER,
        bar_hint_text="Search colors...",
        view_hint_text="Choose a color from the suggestions...",
        controls=[
            ft.ListTile(title=ft.Text(f"Color {i}"), on_click=close_anchor, data=i)
            for i in range(10)
        ],
    )

    page.add(
        ft.Row(
            alignment=ft.MainAxisAlignment.CENTER,
            controls=[
                ft.OutlinedButton(
                    "Open Search View",
                    on_click=lambda _: anchor.open_view(),
                ),
            ],
        ),
        anchor,
    )


ft.app(target=main)

```

**Reply by chaotic-dev** [reply]
This leads to the same issue of the text entry from the `anchor` control not being focused (i.e. no cursor in the search bar for the user to continue typing)
